### ISSUE TITLE
Thunderbird is a lot slower than K9

### ISSUE THREAD

**Issue by zecamigo** [thunderbird/thunderbird-android#8480]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App

Thunderbird for Android

### App version

8.0

### Where did you get the app from?

F-Droid

### Android version

13 (20-20241011-dos-taimen)

### Steps to reproduce

1. Use app.
2. Pick any account on the menu
3. Pick the default folder (usually Inbox)

### Expected behavior

I expected the app to run as smooth or smoother than K9. It was supposed to be an upgrade.

### Actual behavior

Each account/folder touch makes the message list refresh, which takes approximately half a second. 
List is emptied than filled up again each time.
This happens even when you choose an account, and then the default, preselected and loaded folder.

So it unnecessary reloads the default folder's message list.

Also loading messages is a lot slower than K9, and feels glitchy.

EDIT: describing steps to reproduce more clearly

**Comment by cketti** [comment]
The apps use the same code base. So there shouldn't be any differences in speed when the environment is the same.

Loading contact names can be very slow. Did you maybe enable that in Thunderbird, but not K-9 Mail? *Settings → General settings → Display → Show contact names*

Is K-9 Mail configured to only keep a few messages on the local device and Thunderbird is configured to keep a copy of all messages on the server?

**Comment by zecamigo** [comment]
Thanks for replying.

I believe the K9 version I was using was the 6.xxx so there were changes in 7 I did not notice.

K9 then upgraded to TB 8(beta).

It appears there maybe visual effects (like a fading in of messages body?) Even after I disable gaudy visuals.

I did not change any other settings besides this.


But one thing at a time:

- The message list does reload unnecessarily after you pick account> pick default folder.


> Is K-9 Mail configured to only keep a few messages on the local device and Thunderbird is configured to keep a copy of all messages on the server?

I imported settings from K9 so should be the same.
-- 
José Castro

**Comment by zecamigo** [comment]
Another event is when you pick a message from a thread and it slides to the side. 
If this was an already existing effect it was a lot quicker to the point I never noticed it.

Once the message has been viewed (cached?) it becames smoother.

**Comment by zecamigo** [comment]
> Once the message has been viewed (cached?) it becames smoother.

Until it doesn't again.

Isn't it also parsing more HTML elements than K9 6.xxx? This could be another cause.

**Comment by kanashimia** [comment]
Switching between folders seems slower after updating to 8.0 from 6.904
I don't know if this is a real regression, but it should be improved regardless.

It needs to be fixed in multiple ways:
1. Folder switching should be faster, there shouldn't be any visible delay, why does it take so long?
2. Changing views shouldn't cause empty message list to be shown. When you open a message thread it blinks too. Basically it shows empty message list first, and after that fades in actual messages and a thread title. It is better to show current message list and then immediately switch to another one (populated with messages) after it is loaded. It is still possible to have some short fade between views.
3. Some animations are impossible to disable, even if you disable gaudy animations in k9 and disable animations in android settings. I know a person who is extremely annoyed by this to the point of not using k9, seems strange I know.

Current behavior is very annoying for people who don't like flashing lights.

**Comment by Phoenix616** [comment]
> Some animations are impossible to disable, even if you disable gaudy animations in k9 and disable animations in android settings. I know a person who is extremely annoyed by this to the point of not using k9, seems strange I know. (specifically annoyed with drawer animations)

Some more info regarding that point: The app seems to ignore the transition animation scale and the animator duration scale. (Not sure which should apply, I have all of mine set to 0.5 normally and would prefer for this app to respect it too). 

Tested changing it in the android developer options to something insanely high and the app keeps the same speed. Changing the "Window animation scale" however actually does something: It changes the fade in duration of the side drawer (if set to 10 it keeps being transparent for longer? Not sure why that effect is even there but maybe it helps debug this issue)

**Comment by kanashimia** [comment]
I also found that pressing on the account name opens inbox faster that just pressing on the inbox??????? And it doesn't hide the drawer automatically when you press on the account. When you presson the inbox it seems to switch folder only after the drawer hiding animation is done, I don't know if it is a coincidence. Something pretty stupid going on.

**Comment by EchedelleLR** [comment]
I experince the same issues here in comparison to K9 6.x. Even reloading inbox to search new messages is slower by a difference of seconds (6.x) and minutes (8.x).

**Comment by daviewales** [comment]
I have exactly this experience. I was on the pre-thunderbird version of K-9 from FDroid, then one day version 8 appeared as an in-place update. Since then, the app has been extremely show to display the message list, and never shows it at all for the default unified inbox. It's also slow when deleting messages.

**Comment by angerangel** [comment]
I'm using 12.1 with pop3 and it's so slow that I can't use it. I limited the messages to 500, but folder counter still display over 1K (so more than 1000). 
I deleted all messeges with _clear folder_ function, and then it still downloaded over 1000 messages.
I suppose the problem is in the number of messages that are not automatically deleted.

**Comment by Alecaddd** [comment]
These symptoms seems to match the same reported and investigate in #10373 
I'm gonna close it as duplicate since some work is already in progress to fix the issue.

**Comment by EchedelleLR** [comment]
@Alecaddd closing as a duplicate of a newer issue makes really no sense ^^;;

**Comment by EchedelleLR** [comment]
I guess the reporter got a DeLorean and was misplaced temporarily.
